remix-astwalker/src/sourceMappings.ts

```diff
@@ -0,0 +1,105 @@
+import { AstWalker } from './astWalker';
+import { AstNode, Location } from "./index";
+
+/**
+ * Break out fields of an AST's "src" attribute string (s:l:f)
+ * into its "start", "length", and "file index" components.
+ *
+ * @param {AstNode} astNode  - the object to convert.
+ */
+export function sourceLocationFromAstNode(astNode: AstNode): Location | null {
+  if (astNode.src) {
+    var split = astNode.src.split(':')
+    return <Location>{
+      start: parseInt(split[0], 10),
+      length: parseInt(split[1], 10),
+      file: parseInt(split[2], 10)
+    }
+  }
+  return null;
+}
+
+/**
+ * Routines for retrieving AST object(s) using some criteria, usually
+ * includng "src' information.
+ */
+export class SourceMappings {
+
+  readonly source: string;
+  readonly lineBreaks: Array<number>;
+
+  constructor(source: string) {
+    this.source = source;
+    this.lineBreaks = this.getLinebreakPositions();
```

Comment:
`.getLinebreakPositions();` needs a `string`

Reply:
ah I see what happen here.
if `getLinebreakPositions` is called from the outside, there should be a way to update `lineBreaks`

Reply:
i suppose there is no need to make `getLinebreakPositions` public then

Reply:
I have removed this. Let me know if this is sufficient.